frontend/src/plugin/registry.tsx

```diff
@@ -126,4 +138,17 @@ export default class Registry {
   registerDetailsViewSection(sectionName: string, sectionFunc: sectionFunc) {
     store.dispatch(setDetailsView(sectionName, sectionFunc));
   }
+  /**
+   *
+   * @param appLogoGeneratorFunc a logo generator function which returns a react component, this component takes two optional props
+   * logoType which is an enum @enum {LOGO_TYPE} that has two values SMALL and LARGE for screen sizes and the other argument is the theme ```JSX Theme``` interface
```

Comment:
@joaquimrocha I am a little confused are you suggesting to return a component or a function that returns a component? Using a component made more sense because All the state related info can be passed to the component via prop.
cc: @illume